src/Traits/FindMethodTrait.php

```diff
@@ -30,13 +31,34 @@ public function findMethod($method = null)
      * @param string $id Record id
      * @param \Crud\Event\Subject $subject Event subject
      * @return \Cake\ORM\Entity
+     * @throws \Cake\Datasource\Exception\InvalidPrimaryKeyException When $primaryKey has an
+     *      incorrect number of elements.
      */
     protected function _findRecord($id, Subject $subject)
     {
         $repository = $this->_table();
 
+        $key = (array)$repository->primaryKey();
+        $alias = $repository->alias();
+        foreach ($key as $index => $keyname) {
+            $key[$index] = $alias . '.' . $keyname;
+        }
+        $primaryKey = (array)$id;
+        if (count($key) !== count($primaryKey)) {
+            $primaryKey = $primaryKey ?: [null];
+            $primaryKey = array_map(function ($key) {
```

Comment:
I just used the same logic as `Table->get()`: https://github.com/alysson-azevedo/cakephp/blob/master/src/ORM/Table.php#L1164-L1182

That array map just adds a ' in each value of the array. A `["key"]` becomes `["'key'"]`.